src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -256,14 +256,11 @@ private boolean quickSetSolid(final TransactionViewModel transactionViewModel) t
     }
 
     private boolean checkApproovee(TransactionViewModel approovee) throws Exception {
-        if(approovee.getType() == PREFILLED_SLOT) {
+        if (approovee.getType() == PREFILLED_SLOT) {
             transactionRequester.requestTransaction(approovee.getHash(), false);
             return false;
         }
-        if(approovee.getHash().equals(Hash.NULL_HASH)) {
-            return true;
-        }
-        return approovee.isSolid();
+        return approovee.getHash().equals(Hash.NULL_HASH) || approovee.isSolid();
```

Comment:
Even though `approovee.getHash()` will never be null,
please put Hash.NULL_HASH first or use `Objects.equals()`